src/components/editor/EditAuthorList.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import _ from 'lodash';
+import { shallowEquals } from 'lib/BaseComponent';
+
+export default class EditAuthorList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      // Key is slug, value is name
+      authorsAdded: {},
+      // An object where key is slug and value is whether it's deleted or not
+      authorsDeleted: this.props.data.reduce((cur, next) => {
+        cur[next.slug] = false;
+        return cur;
+      }, {})
+    };
+    this.handleAddAuthor = this.handleAddAuthor.bind(this);
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
```

Comment:
I actually don't quite get the shouldComponentUpdate in basecomponent, why is it we only update when state didn't change and props did? Why don't we update when state changes?